File: IntegrationCore/Models/DB/Integration.cs
using System.Collections.Generic;

namespace IntegrationCore.Models.DB
{
    public class Integration
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public IEnumerable<ConnectionFieldValue> ConnectionFieldValues { get; set; }
        public IEnumerable<FieldConnection> FieldConnections { get; set; }
        public int TypeFromId { get; set; }
        public TypeDefinition TypeFrom { get; set; }
        public int TypeToId { get; set; }
        public TypeDefinition TypeTo { get; set; }
    }
}

File: IntegrationCore/ClientApp/src/app/models/integration.ts
import { TypeDefinition } from './type-definition';
import { ConnectionFieldValue } from './connection-field-value';
import { FieldConnection } from './field-connection';

export class Integration {
  id: number;
  name: string;
  connectionFieldValues: ConnectionFieldValue[];
  fieldConnections: FieldConnection[];
  typeFromId: number;
  typeFrom: TypeDefinition;
  typeToId: number;
  typeTo: TypeDefinition;
}
